htlcswitch/link.go

```diff
@@ -2648,12 +2648,17 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 		fwdInfo, err := chanIterator.ForwardingInstructions()
 		if err != nil {
 			// If we're unable to process the onion payload, or we
-			// we received malformed TLV stream, then we should
-			// send an error back to the caller so the HTLC can be
-			// canceled.
+			// we received invalid onion payload failure, then we
+			// should send an error back to the caller so the HTLC
+			// can be cancelled.
+			var failedType uint64
+			if e, ok := err.(hop.ErrInvalidPayload); ok {
+				failedType = uint64(e.Type)
+			}
+
 			l.sendHTLCError(
 				pd.HtlcIndex,
-				lnwire.NewInvalidOnionVersion(onionBlob[:]),
+				lnwire.NewInvalidOnionPayload(failedType, 0),
```

Comment:
Maybe document offset zero in the code too?

Reply:
Add the TODO in the code as well? A todo in a commit message, is it ever going to be seen again?

Reply:
moved todo into file